<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SimpleHasherMainWindow</class>
 <widget class="QMainWindow" name="SimpleHasherMainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>716</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Simple Hasher</string>
  </property>
  <property name="windowIcon">
   <iconset resource="rsc/resources.qrc">
    <normaloff>:/SimpleHasher/application.svg</normaloff>:/SimpleHasher/application.svg</iconset>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout" stretch="1,0,0,0">
    <item>
     <widget class="QTableWidget" name="m_hashTable">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="toolTip">
       <string>List of files and computed hashes</string>
      </property>
      <property name="sizeAdjustPolicy">
       <enum>QAbstractScrollArea::SizeAdjustPolicy::AdjustToContents</enum>
      </property>
      <property name="alternatingRowColors">
       <bool>true</bool>
      </property>
      <property name="columnCount">
       <number>0</number>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="m_hashGroup">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="styleSheet">
       <string notr="true">QGroupBox {
    border: 1px solid gray;
    border-radius: 5px;
    margin-top: 2ex;
}

QGroupBox::title {
    subcontrol-origin: margin;
    subcontrol-position: top center; /* position at the top center */
    padding: 0px 5px;
}</string>
      </property>
      <property name="title">
       <string>Hash Algorithms</string>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <widget class="QCheckBox" name="m_md5">
         <property name="toolTip">
          <string>Message-Digest Algorithm 5 (128 bits)</string>
         </property>
         <property name="text">
          <string>MD5</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="m_sha1">
         <property name="toolTip">
          <string>Secure Hash Algorithm 1 (160 bits)</string>
         </property>
         <property name="text">
          <string>SHA-1</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="m_sha224">
         <property name="toolTip">
          <string>Secure Hash Algorithm 2 (224 bits)</string>
         </property>
         <property name="text">
          <string>SHA-224</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="m_sha256">
         <property name="toolTip">
          <string>Secure Hash Algorithm 2 (256 bits)</string>
         </property>
         <property name="text">
          <string>SHA-256</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="m_sha384">
         <property name="toolTip">
          <string>Secure Hash Algorithm 2 (384 bits)</string>
         </property>
         <property name="text">
          <string>SHA-384</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="m_sha512">
         <property name="toolTip">
          <string>Secure Hash Algorithm 2 (512 bits)</string>
         </property>
         <property name="text">
          <string>SHA-512</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="m_tiger">
         <property name="toolTip">
          <string>Tiger (192 bits)</string>
         </property>
         <property name="text">
          <string>Tiger</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer_2">
         <property name="orientation">
          <enum>Qt::Orientation::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_4">
      <item>
       <widget class="QProgressBar" name="m_progress">
        <property name="font">
         <font>
          <bold>true</bold>
         </font>
        </property>
        <property name="toolTip">
         <string>Current process progress.</string>
        </property>
        <property name="value">
         <number>24</number>
        </property>
        <property name="alignment">
         <set>Qt::AlignmentFlag::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="m_cancel">
        <property name="minimumSize">
         <size>
          <width>24</width>
          <height>24</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>24</width>
          <height>24</height>
         </size>
        </property>
        <property name="toolTip">
         <string>Cancel current process.</string>
        </property>
        <property name="text">
         <string>...</string>
        </property>
        <property name="icon">
         <iconset resource="rsc/resources.qrc">
          <normaloff>:/SimpleHasher/cancel.svg</normaloff>:/SimpleHasher/cancel.svg</iconset>
        </property>
        <property name="iconSize">
         <size>
          <width>20</width>
          <height>20</height>
         </size>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <widget class="QPushButton" name="m_addFile">
        <property name="toolTip">
         <string>Add files to the list to hashed</string>
        </property>
        <property name="statusTip">
         <string>Add files to the list to hashed</string>
        </property>
        <property name="text">
         <string>Add</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="m_removeFile">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="toolTip">
         <string>Remove data from the list to be hashed</string>
        </property>
        <property name="statusTip">
         <string>Remove data from the list to be hashed</string>
        </property>
        <property name="text">
         <string>Remove</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="m_compute">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="toolTip">
         <string>Compute selected hashes for the group of files</string>
        </property>
        <property name="statusTip">
         <string>Compute selected hashes for the group of files</string>
        </property>
        <property name="text">
         <string>Compute</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="m_save">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="toolTip">
         <string>Save hashes to a file</string>
        </property>
        <property name="statusTip">
         <string>Save hashes to a file</string>
        </property>
        <property name="text">
         <string>Save</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Orientation::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="m_options">
        <property name="text">
         <string>Options...</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="m_about">
        <property name="toolTip">
         <string>About Simple Hasher</string>
        </property>
        <property name="statusTip">
         <string>About Simple Hasher</string>
        </property>
        <property name="text">
         <string>About...</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="m_quit">
        <property name="toolTip">
         <string>Quit the application</string>
        </property>
        <property name="statusTip">
         <string>Quit the application</string>
        </property>
        <property name="text">
         <string>Quit</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <resources>
  <include location="rsc/resources.qrc"/>
 </resources>
 <connections/>
</ui>
